### ISSUE TITLE
non-blocking errors : UnhandledPromiseRejectionWarning & DeprecationWarning

### ISSUE THREAD

**Issue by joelgrimberg** [adamgruber/mochawesome#139]
when using Mochawesome within my Codeceptjs-tests,
I get the following errors:
(node:7909) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: exit is not a function
(node:7909) DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

## Node version:
7.7.1

## NPM version:
4.1.2
## Mocha version:
mocha@3.2.0

## Mochawesome version:
mochawesome@2.0.4

reporteroptions:
"mocha": {
    "reporterOptions": {
      "mochaFile": "output/result.xml",
      "reportDir": "output",
      "reportFilename": "check",
      "enableCharts": true,
      "autoOpen": true,
      "quiet": true
    }
  },

**Comment by adamgruber** [comment]
I haven't seen this error in any of my tests. Can you provide a sample setup with code that will reproduce this?

**Comment by joeypedicini92** [comment]
I'm getting this as well. 

Node: v6.9.1
Codeceptjs: v0.5.1
Mochawesome: v2.0.4
Nightmare: v2.10.0

```
[mochawesome] Report JSON saved to xxx/mochawesome.json

[mochawesome] Report HTML saved to xxx/mochawesome.html

[mochawesome] {}

(node:78786) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): TypeError: exit is not a function
```

**Comment by adamgruber** [comment]
This was addressed in version 2.0.5. Try upgrading and let me know if the issue still persists.

**Comment by fincha** [comment]
```
(node:45324) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 52): ValidationError: AuditEntry validation failed
(node:45324) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 53): ValidationError: AuditEntry validation failed
(node:45324) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 54): ValidationError: AuditEntry validation failed
(node:45324) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 59): ValidationError: AuditEntry validation failed
(node:45324) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 60): ValidationError: AuditEntry validation failed
(node:45324) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 61): ValidationError: AuditEntry validation failed
(node:45324) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 62): ValidationError: AuditEntry validation failed
(node:45324) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 67): ValidationError: AuditEntry validation failed
```
